deployment/data-streams/changeset/deploy.go

```diff
@@ -0,0 +1,62 @@
+package changeset
+
+import (
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+)
+
+type (
+	// Contract covers contracts such as channel_config_store.ChannelConfigStore and fee_manager.FeeManager.
+	Contract interface {
+		// Caller:
+		Owner(opts *bind.CallOpts) (common.Address, error)
+		SupportsInterface(opts *bind.CallOpts, interfaceID [4]byte) (bool, error)
+		TypeAndVersion(opts *bind.CallOpts) (string, error)
+
+		// Transactor:
+		AcceptOwnership(opts *bind.TransactOpts) (*types.Transaction, error)
+		TransferOwnership(opts *bind.TransactOpts, to common.Address) (*types.Transaction, error)
+	}
+
+	ContractDeployFn[C Contract] func(chain deployment.Chain) *ContractDeployment[C]
+
+	ContractDeployment[C Contract] struct {
+		Address  common.Address
+		Contract C
+		Tx       *types.Transaction
+		Tv       deployment.TypeAndVersion
+		Err      error
+	}
+)
+
+var _ deployment.ChangeSet[DeployChannelConfigStoreConfig] = DeployChannelConfigStore
```

Comment:
should we be using the newer `deployment.ChangeSetV2` or the `deployment.ChangeLogic` since deployment.ChangeSet is deprecated an this is still in progress